src/Levenshtein.elm

```diff
@@ -18,51 +18,54 @@ distance str1 str2 =
         0
 
     else
-        helper
+        distanceHelper
             (Array.fromList (String.toList str1))
             (Array.fromList (String.toList str2))
 
 
-helper arr1 arr2 =
+distanceHelper : Array Char -> Array Char -> Int
+distanceHelper arr1 arr2 =
     let
-        lev : Table -> ( Int, Int ) -> ( Table, Int )
-        lev table ( i, j ) =
+        calculateEditDistanceForChars : Table -> ( Int, Int ) -> ( Table, Int )
+        calculateEditDistanceForChars table ( i, j ) =
             case ( Array.get (i - 1) arr1, Array.get (j - 1) arr2 ) of
                 ( Just chr1, Just chr2 ) ->
                     let
-                        indicator =
-                            if chr1 /= chr2 then
-                                1
-
-                            else
-                                0
-
                         ( table1, dist1 ) =
-                            Table.fetch ( i - 1, j ) lev table
+                            Table.fetch ( i - 1, j ) calculateEditDistanceForChars table
 
                         ( table2, dist2 ) =
-                            Table.fetch ( i, j - 1 ) lev table1
+                            Table.fetch ( i, j - 1 ) calculateEditDistanceForChars table1
 
                         ( table3, dist3 ) =
-                            Table.fetch ( i - 1, j - 1 ) lev table2
+                            Table.fetch ( i - 1, j - 1 ) calculateEditDistanceForChars table2
                     in
                     ( table3
-                    , min3
-                        (dist1 + 1)
-                        (dist2 + 1)
-                        (dist3 + indicator)
+                      -- A more optimized version of
+                      -- min (min (dist1 + 1) (dist2 + 1)) (dist3 + indicator)
+                    , if dist3 < dist1 then
```

Comment:
This is the perfomance change. The calls to `min3` and `min` have been inlined, such that the `indicator` variable can be calculated only when it's needed. Though the calculation is simple enough on it's own, it involves a branch, which can throw a wrench into any hot code.